### PR TITLE
Extendable 'SetupWallet' method. 

### PR DESCRIPTION
**Pull request by codingupastorm**
Need to add functionality to this method in other PRs to write more tests for the WalletTransactionHandler. Not keen on adding to that multiple return.

### CODE REVIEW

File: src/Stratis.Bitcoin.Features.Wallet.Tests/WalletTransactionHandlerTest.cs [hunk 1]
```diff
@@ -148,7 +148,13 @@ public void BuildTransactionFeeTooLowThrowsWalletException()
         [Fact]
         public void BuildTransactionNoChangeAdressesLeftCreatesNewChangeAddress()
         {
-            (Wallet wallet, (ExtKey ExtKey, string ExtPubKey) accountKeys, (PubKey PubKey, BitcoinPubKeyAddress Address) destinationKeys, TransactionData addressTransaction, WalletTransactionHandler walletTransactionHandler, WalletAccountReference walletReference) = this.SetupWallet();
+            WalletTransactionHandlerTestContext testContext = SetupWallet();
+            Wallet wallet = testContext.Wallet;
```
Comment (fassadlr) [inline src/Stratis.Bitcoin.Features.Wallet.Tests/WalletTransactionHandlerTest.cs]: @codingupastorm why create variables in each test when you can just pass the `WalletTransactionHandlerTestContext` instance to `CreateContext`?

Comment (fassadlr) [reply src/Stratis.Bitcoin.Features.Wallet.Tests/WalletTransactionHandlerTest.cs]: Or whatever is required...

Comment (codingupastorm) [reply src/Stratis.Bitcoin.Features.Wallet.Tests/WalletTransactionHandlerTest.cs]: Ah, in the future we may have other params to add to CreateContext, so it doesn't really make sense. 

I'm thinking this after working on the TransactionBuildOptions coming in a new PR (hopefully tomorrow!)

Comment (fassadlr) [reply src/Stratis.Bitcoin.Features.Wallet.Tests/WalletTransactionHandlerTest.cs]: What I mean is, can this test be written like this?

```
    WalletTransactionHandlerTestContext testContext = SetupWallet();
    Wallet wallet = testContext.Wallet;
    (ExtKey ExtKey, string ExtPubKey) accountKeys = testContext.AccountKeys;
    WalletTransactionHandler walletTransactionHandler = testContext.WalletTransactionHandler;
    TransactionData addressTransaction = testContext.AddressTransaction;

    TransactionBuildContext context = CreateContext(this.Network, testContext.WalletReference, "password", testContext. DestinationKeys.PubKey.ScriptPubKey, new Money(7500), FeeType.Low, 0);
    ...

```

etcetc

Comment (codingupastorm) [reply src/Stratis.Bitcoin.Features.Wallet.Tests/WalletTransactionHandlerTest.cs]: Sure, I can do that. I was very lazy

Comment (codingupastorm) [reply src/Stratis.Bitcoin.Features.Wallet.Tests/WalletTransactionHandlerTest.cs]: Done :)

**Comment by fassadlr** [comment]
@codingupastorm Can you check please, I don't the solution builds?

**Pull request merged**